src/Scrutor/ServiceCollectionExtensions.Decoration.cs

```diff
@@ -282,16 +283,15 @@ private static bool TryDecorateDescriptors(this IServiceCollection services, Typ
             {
                 return false;
             }
+            
+            services.TryAddTransient<DecoratedServiceDisposer>();
```

Comment:
Registering and resolving transients from the top-level provider is almost guaranteed a memory leak... 🙈

Reply:
Resolution of `DecoratedServiceDisposer` from root happens 
1. for singletons only and that's correct because singletons are resolved from root as well.
2. for non-singleton that are explicitly resolved from the root

In both cases the decoratee, the decorator and `DecoratedServiceDisposer` are captured by the root scope and must be disposed of by the root (only).

Can you provide an example so I understand your concerns?